### PR TITLE
YLCI: Merfolk Tunnel-Guide and support (X-time boons)

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-gui/res/cardsfolder/a/arcane_archery.txt [hunk 1]
```diff
@@ -2,12 +2,11 @@ Name:Arcane Archery
...
-SVar:SpellCast:Mode$ SpellCast | ValidCard$ Creature | ValidActivatingPlayer$ You | OneOff$ True | ActivationLimit$ 1 | TriggerZones$ Command | Execute$ ReplEffAddCounter | TriggerDescription$ When you cast your next creature spell, that creature enters the battlefield with an additional +1/+1 counter, reach counter, and trample counter on it.
+SVar:DBBoon:DB$ Effect | Boon$ True | Duration$ Permanent | Triggers$ SpellCast | SpellDescription$ ,,,,,,You get a boon with "When you cast your next creature spell, that creature enters the battlefield with an additional +1/+1 counter, reach counter, and trample counter on it."
+SVar:SpellCast:Mode$ SpellCast | ValidCard$ Creature | ValidActivatingPlayer$ You | TriggerZones$ Command | Execute$ ReplEffAddCounter | TriggerDescription$ When you cast your next creature spell, that creature enters the battlefield with an additional +1/+1 counter, reach counter, and trample counter on it.
 SVar:ReplEffAddCounter:DB$ Effect | ReplacementEffects$ ETBAddCounter | RememberObjects$ TriggeredCard
 SVar:ETBAddCounter:Event$ Moved | Origin$ Stack | Destination$ Battlefield | ValidCard$ Card.IsRemembered | ReplaceWith$ ETBAddExtraCounter
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/a/arcane_archery.txt:8]: most likely broken now without Updated result

File: forge-gui/res/cardsfolder/c/chancellor_of_the_annex.txt [hunk 2]
```diff
@@ -5,9 +5,9 @@ PT:5/6
 K:Flying
 K:MayEffectFromOpeningHand:RevealCard
 SVar:RevealCard:DB$ Reveal | RevealDefined$ Self | SubAbility$ CounterFirstSpellEffect | SpellDescription$ You may reveal this card from your opening hand. If you do, when each opponent casts their first spell of the game, counter that spell unless that player pays {1}.
-SVar:CounterFirstSpellEffect:DB$ RepeatEach | RepeatPlayers$ Player.Opponent | RepeatSubAbility$ DBEffect
-SVar:DBEffect:DB$ Effect | Triggers$ TrigCounter | EffectOwner$ Player.IsRemembered | Duration$ Permanent
-SVar:TrigCounter:Mode$ SpellCast | ValidCard$ Card | ValidActivatingPlayer$ You | Execute$ TrigCounterSpell | OneOff$ True | TriggerZones$ Command | TriggerDescription$ Whenever an opponent casts a spell for the first time, counter that spell unless that player pays {1}.
+SVar:CounterFirstSpellEffect:DB$ RepeatEach | RepeatPlayers$ Opponent | RepeatSubAbility$ DBEffect
```
[the remark this answers is not in the document]
Comment (Northmoc) [reply forge-gui/res/cardsfolder/c/chancellor_of_the_annex.txt:8]: I think probably to handle this. The Oracle wording is not very clear
image

File: forge-gui/res/cardsfolder/a/arcane_archery.txt (hunk 1 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/a/arcane_archery.txt:8]: Hmm, initial testing of other cards seemed good... will check

File: forge-gui/res/cardsfolder/c/chancellor_of_the_annex.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (tool4ever) [reply forge-gui/res/cardsfolder/c/chancellor_of_the_annex.txt:8]: sure, but what about `EffectOwner$ Opponent`?

File: forge-gui/res/cardsfolder/a/arcane_archery.txt (hunk 1 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/a/arcane_archery.txt:8]: Good catch

File: forge-game/src/main/java/forge/game/trigger/TriggerHandler.java [hunk 3]
```diff
@@ -563,7 +563,13 @@ else if (sa instanceof AbilitySub || !sa.hasParam("Cost") || (sa.getPayCosts() !
 
         regtrig.triggerRun();
 
-        if (regtrig.hasParam("OneOff") && host.isImmutable()) {
+        boolean removeBoon = host.isBoon();
+        if (regtrig.hasParam("BoonAmount")) {
+                int x = AbilityUtils.calculateAmount(host, regtrig.getParam("BoonAmount"), wrapperAbility);
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/trigger/TriggerHandler.java:568]: formatting seems off

**Review by tool4ever — APPROVED** [review]

**Pull request merged**